test-harness/index.js

```diff
@@ -0,0 +1,265 @@
+const { join } = require('path');
+const fetch = require('node-fetch');
+const requests = require('./requests');
+
+const { exec } = require('child_process');
+const { getPort, makeRequest, constructMasterFileName, readFile } = require('./helpers');
+
+const port = getPort(process);
+
+async function waitForPrism(done) {
+  try {
+    await fetch(`http://localhost:${port}`);
+    done();
+  } catch (err) {
+    setTimeout(() => {
+      waitForPrism(done);
+    }, 500);
+  }
+}
+
+async function runTest(req) {
+  const { dynamic, ...request } = req;
+
+  const masterFileName = constructMasterFileName(request);
+  const masterFile = readFile(masterFileName);
+
+  const reqRes = await makeRequest(request);
+
+  return {
+    reqRes,
+    masterFile,
+  };
+}
+
+const spec = process.env.SPEC || join(__dirname, '/../examples/petstore.oas2.json');
+const specs = spec.split(',');
+
+const createSpec = (specPath, prismCmd) => {
+  return () => {
+    describe(specPath, () => {
+      let prismProcessHandle;
```

Comment:
@XVincentX, the previous version didn't need any `let`. I don't see how is this and later mutation with `prismProcessHandle = exec(prismCmd);` better? I'd rather go back 2 commits.

Reply:
Let me finish first :)